src/node_crypto.cc

```diff
@@ -3583,7 +3583,7 @@ SignBase::Error Sign::SignFinal(const char* key_pem,
   if (pkey == nullptr)
     goto exit;
 
-  if (EVP_SignFinal(&mdctx_, *sig, sig_len, pkey))
+  if (EVP_SignFinal(&mdctx_, *sig, sig_len, pkey) && 0 == ERR_peek_error())
```

Comment:
I just did some debugging and the error itself happens in `PEM_read_bio_PrivateKey`. May I ask you to move the `ERR_peek_error()` check there along with `pkey == nullptr`?

A note with explanation about the weird behavior of `PEM_read_bio_PrivateKey` would be twice as welcome!